staging/src/k8s.io/apiserver/pkg/server/options/audit.go

```diff
@@ -250,44 +272,88 @@ func (o *AuditOptions) AddFlags(fs *pflag.FlagSet) {
 	o.WebhookOptions.AddFlags(fs)
 	o.WebhookOptions.BatchOptions.AddFlags(pluginwebhook.PluginName, fs)
 	o.WebhookOptions.TruncateOptions.AddFlags(pluginwebhook.PluginName, fs)
+	o.DynamicOptions.AddFlags(fs)
 }
 
-func (o *AuditOptions) ApplyTo(c *server.Config) error {
+func (o *AuditOptions) ApplyTo(
+	c *server.Config,
+	kubeClientConfig *restclient.Config,
+	informers informers.SharedInformerFactory,
+	processInfo *ProcessInfo,
+	serviceResolver utilwebhook.ServiceResolver,
+	authInfoResolverWrapper utilwebhook.AuthenticationInfoResolverWrapper,
+) error {
 	if o == nil {
 		return nil
 	}
+	if c == nil {
+		return fmt.Errorf("server config must be non-nil")
+	}
 
 	// Apply advanced options.
 	// 1. Apply generic options.
-	if err := o.applyTo(c); err != nil {
+	checker, err := o.applyTo(*c)
+	if err != nil {
 		return err
 	}
+	c.AuditPolicyChecker = checker
```

Comment:
this reads really oddly... `applyTo(c)` doesn't modify `c`, but it returns something we then set into `c`, and then conditionally override later?

Reply:
This is because the webhook and log `applyTo` methods need to check if a policy exists, and bail with a relevent error message if it doesn't. The checker is explicitly returned so that it can be overridden if the dynamic policy exists. Stefan had asked that we not modify `c` and return things in methods.  I could optionally pass the checker as a param if thats simpler.